### ISSUE TITLE
Danger inline commenting doesn't seem to work with space in path

### ISSUE THREAD

**Issue by freak4pc** [danger/danger#1042]
# Report

## What did you do?

I'm using danger-ruby-swiftlint, using inline_mode which just dispatches the warn/fail messages to danger with the file path. It seems, though, that if the path contains a space, the message isn't dispatched as inline, but fall-backs to a regular comment.

I've [opened an issue on that repo](https://github.com/ashfurrow/danger-ruby-swiftlint/issues/115), but after digging a bit I feel quite confident the issue is inside danger: 

Unfortunately as much as I've tried, I don't have enough Ruby knowledge, or danger-specific knowledge, to understand where and how to start debugging this. 

Working example:
![image](https://user-images.githubusercontent.com/605076/49005262-f130bd80-f16e-11e8-9c69-2a8bcb7ef521.png)

Test where filename has space just fall-backs into a regular warning:
![image](https://user-images.githubusercontent.com/605076/49005284-01489d00-f16f-11e8-8393-b9e36c2c06a2.png)

## What did you expect to happen?

All inline comments are posted inline with the provided file path. 

## What happened instead?

If a file path has a space, instead of inline comment, it falls back to regular comment: 

## Your Environment

* Which CI are you running on? Bitrise
* Are you running the latest version of Danger? Yes
* What is your Dangerfile? for the case of this just `swiftlint.lint` with inline mode.

**Comment by freak4pc** [comment]
If anyone knows enough Ruby or knows this repo enough to at least direct me to the right area, I'd love take a crack at this! My attempt at this was unsuccessful, to say the last :D

**Comment by orta** [comment]
Try looking at the tests for inline issues, and see if you can re-create the failing case, from there you can reliably break stuff till it works, at a guess, maybe an encoding issue?

https://github.com/danger/danger/blob/master/spec/lib/danger/request_sources/github_spec.rb#L365-L513

**Comment by freak4pc** [comment]
Yeah that's what I already played with but I'm just not fluent with Ruby Specs, Fixtures, and all that jazz. Couldn't really get it to feel. Any help from anyone in this community would be highly appreciated 🙏

**Comment by colorbox** [comment]
I think this issue has solved.

I research about this problem and find out that this problem derives from GitHub API.
I posted a question on GitHub community and get answered that problem has solved.

https://github.community/t5/GitHub-API-Development-and/How-can-I-create-PR-comment-on-the-file-that-includes-space/m-p/16503

After getting answer, My PR to test this problem has been solved.

@freak4pc  How about your PR on your repository?
Is this issue solved?

**Comment by daniel-beard** [comment]
I still see this issue with v5.15.0, inline commenting doesn't appear to work at all for me when running a shared danger file imported via git.

**Comment by daniel-beard** [comment]
The referenced PRs should have fixed this issue, I think it can be closed now.

**Comment by freak4pc** [comment]
I don't have the same use case anymore so can't really test it. Feel free to close if it works for you :)

**Comment by eneko** [comment]
This issue might be related, I can open a new one if needed.

It seems Danger has trouble with file paths that contain spaces in them. This issue is still present on Danger 6.1.0.

In my case, Danger fails with an **"Invalid `Dangerfile` file: invalid byte sequence in UTF-8"** when the git diff contains files with spaces in their file path (some files with spaces on them are being deleted from git in the log below).

```
Using danger 6.1.0
357Using danger-plugin-api 1.0.0
358Using danger-auto_label 1.3.1
359Using thor 0.20.3
360Using danger-swiftlint 0.23.0
361Bundle complete! 3 Gemfile dependencies, 26 gems now installed.
362Bundled gems are installed into `./vendor/bundle`
363The command "bundle install" exited with 0.
3643.72s$ bundle exec danger
365NOTE: Inheriting Faraday::Error::ClientError is deprecated; use Faraday::ClientError instead. It will be removed in or after version 1.0
366Faraday::Error::ClientError.inherited called from /Users/travis/build/xyz/vendor/bundle/ruby/2.6.0/gems/octokit-4.14.0/lib/octokit/middleware/follow_redirects.rb:14.
367/Users/travis/build/xyz/vendor/bundle/ruby/2.6.0/gems/git-1.5.0/lib/git/diff.rb:135:in `split':  (Danger::DSLError)
368[!] Invalid `Dangerfile` file: invalid byte sequence in UTF-8
369 #  from Dangerfile:3
370 #  -------------------------------------------
371 #  
372 >  touched_files = git.modified_files + git.added_files
373 #  
374 #  -------------------------------------------
```

The error does not happen when files affected by the diff do not contain spaces.

**Comment by daniel-beard** [comment]
@eneko this appears to be caused by this issue in octokit (which STILL hasn't been merged yet):
https://github.com/octokit/octokit.rb/pull/1118

I use the following as a workaround:

```ruby
require 'git_diff_parser'

# Returns the GitDiffParser::Patches for the current pr
# Cached, so this method can be called multiple times without penalty.
def git_diff_patches()
  # Need to force encoding because of https://github.com/octokit/octokit.rb/pull/1118
  $_pr_diff ||= github.pr_diff.force_encoding('UTF-8')
  $_git_diff_patches ||= GitDiffParser::Patches.parse($_pr_diff)
end
```

**Comment by eneko** [comment]
Thank you, @daniel-beard, will check it out.

**Comment by rahul-narkhede** [comment]
Can you provide an example of how to send an inline message? I have been doing something like 
```
message("This is test message", file: Path/to/file, line: 80)
```
And I get the message posted on PR instead of an inline comment on the file in the PR.  The message on PR has link `path/to/file#80`.
